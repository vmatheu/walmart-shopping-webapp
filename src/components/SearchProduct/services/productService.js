import axios from 'axios';
import { getEnvironment } from 'core/environment';
import { PRODUCT_SERVICE_ENDPOINT } from '../constants';

export const findProductBySearch = async (search) => {
  const productServiceEnpoint = PRODUCT_SERVICE_ENDPOINT[getEnvironment()].url;
  const data = await axios.get(`${productServiceEnpoint}${search}`)
    .then(response => (response.data))
    .catch(error => ([]));
  return data;
};

export default { findProductBySearch };
