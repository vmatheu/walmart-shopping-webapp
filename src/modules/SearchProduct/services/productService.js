import Logger from 'core/services/Logger';
import { PRODUCT_SERVICE_ENDPOINT } from '../constants';

const getProductById = (productId) => {
  Logger.debbug(`product id for get endpoint is ${productId}`);
  Logger.debbug(`Url calling is ${PRODUCT_SERVICE_ENDPOINT}`);

  return [];
};

export default { getProductById };
