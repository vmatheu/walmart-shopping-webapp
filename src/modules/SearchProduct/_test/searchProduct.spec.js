import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';
import productService from '../services/productService';
import { SearchProduct, onChange } from '..';

jest.mock('../services/productService');

describe('<SearchProduct/>', () => {

  describe('With search product by id', () => {
    it('should have the same ui structure', () => {
      const wrapper = shallow(<SearchProduct />);
      const tree = toJson(wrapper);
      expect(tree).toMatchSnapshot();
    });
  });

  describe('Onchange action', () => {
    beforeEach(() => {
      productService.getProductById = jest.fn();
    });

    it('should calling product service', async () => {
      await onChange(111, jest.fn());
      expect(productService.getProductById).toHaveBeenCalled();
    });

    it('should not calling product service when search contains not alphanumeric characters', async () => {
      await onChange('21-', jest.fn());
      expect(productService.getProductById).not.toHaveBeenCalled();
    });
  });
});
