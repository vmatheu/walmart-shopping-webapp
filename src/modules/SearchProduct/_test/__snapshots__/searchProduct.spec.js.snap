// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchProduct/> With search product by id should have the same ui structure 1`] = `
<div
  className="container"
>
  <div
    className="row col-sm-12 col-md-5 "
  >
    <p>
      Ingresar busqueda de Productos
    </p>
  </div>
  <div
    className="row col-sm-12 col-md-5"
  >
    <input
      className="form-control"
      id="inputSearchById"
      style={
        Object {
          "width": "100%",
        }
      }
    />
  </div>
  <div
    className="container"
    style={
      Object {
        "marginTop": "20px",
      }
    }
  />
</div>
`;
